File: src/main/java/util/controls/AnalogInputCommand.java
package util.controls;

import edu.wpi.first.wpilibj2.command.CommandBase;

public abstract class AnalogInputCommand extends CommandBase {

  public Controller controller;

  @Override
  public void execute() {
    analogExecute(this.controller.getControllerValues());
  }

  public abstract void analogExecute(ControllerValues controllerValues);
}

File: src/main/java/util/AnalogInputCommand.kt
package util

import edu.wpi.first.wpilibj2.command.CommandBase

abstract class AnalogInputCommand : CommandBase() {

    internal lateinit var controller: Controller
    private var firstRun: Boolean = true

    override fun execute() = analogExecute(
        this.controller.getControllerValues()
    )

    abstract fun analogExecute(controllerValues: ControllerValues?)
}
